### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by necessarily-equal**
This PR backports small fixes from me and freem. The "fix missing headers" commit is especially useful as I needed it to be able to compile Daemon.

### CODE REVIEW

File: src/engine/sys/con_tty.cpp [hunk 1]
```diff
@@ -85,9 +85,9 @@ void WriteToStdout(const char* text) {
 
 	if (ret < 0) {
 		// Do not log to stdout
-		Log::Dispatch( Str::Format("Error writing to the terminal: %d", errno),
+		Log::Dispatch( Str::Format("Error writing to the terminal: %s", strerror(errno)),
```
Comment (DolceTriade) [inline src/engine/sys/con_tty.cpp:88]: let's print the code and the string?

**Review by DolceTriade — COMMENTED** [review]
otherwise lgtm

File: src/common/cm/cm_load.cpp [hunk 2]
```diff
@@ -771,7 +771,7 @@ void CM_LoadMap(Str::StringRef name)
 	std::error_code err;
 	std::string mapData = FS::PakPath::ReadFile(mapFile, err);
 	if (err) {
-		Sys::Drop("Could not load %s", mapFile.c_str());
+		Sys::Drop("Could not load %s: %s (code:%02x)", mapFile.c_str(), err.message(), err.value() );
```
Comment (slipher) [inline src/common/cm/cm_load.cpp:774]: The use of an unprefixed hexadecimal number is misleading. If it had only digits 0-9, you couldn't tell it's supposed to be hexadecimal.

File: src/common/Assert.h [hunk 3]
```diff
@@ -31,6 +31,12 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef COMMON_ASSERT_H_
 #define COMMON_ASSERT_H_
 
+#include <utility>
```
Comment (slipher) [inline src/common/Assert.h:34]: Our guidance on the common headers for now is they are not actually expected to be self-contained. What you are supposed to to is always include `Common.h` first in every .cpp file.

File: src/engine/sys/con_tty.cpp (hunk 1 above)
Comment (necessarily-equal) [reply src/engine/sys/con_tty.cpp:88]: The standard in Daemon code seems to be to print only `strerror(errno)`, and not the number. I guess even worst case if the message is translated it's often clear which one it is. If that's a problem though, then we should rework more than just this line

File: src/common/cm/cm_load.cpp (hunk 2 above)
Comment (necessarily-equal) [reply src/common/cm/cm_load.cpp:774]: Good point, fixed

File: src/common/Assert.h (hunk 3 above)
Comment (necessarily-equal) [reply src/common/Assert.h]: I've tried to reproduce the bug to try to find an alternative solution. But somehow, the bug doesn't trigger anymore. Do you have suggestions on how to reproduce again?

In the meantime, I removed the commit so be able to merge this faster. Will merge tomorrow most likely

**Comment by slipher** [comment]
LGTM

**Pull request merged**